problem-based-packs/insecure-transport/go-stdlib/http-request.yaml

```diff
@@ -4,10 +4,16 @@ rules:
       - pattern-either:
           - pattern: |
               http.$FUNC("=~/[hH][tT][tT][pP]://.*/", ...)
-          - pattern: |
-              $CLIENT := &http.Client{...}
-              ...
-              client.$FUNC("=~/[hH][tT][tT][pP]://.*/", ...)
+          - patterns:
+            - pattern-inside: |
+                $CLIENT := &http.Client{...}
+                ...
+            - pattern: |
+                client.$FUNC("=~/[hH][tT][tT][pP]://.*/", ...)
+      - pattern-not:
+          http.$FUNC("=~/[hH][tT][tT][pP]://127.0.0.1.*/", ...)
+      - pattern-not:
+          client.$FUNC("=~/[hH][tT][tT][pP]://127.0.0.1.*/", ...)
```

Comment:
might also be helpful to ignore `http://localhost/` but we can do that as a future enhancement if necessary.